pubtools/pulplib/_impl/client/client.py

```diff
@@ -775,7 +775,7 @@ def _do_associate(self, src_repo_id, dest_repo_id, criteria=None, raw_options=No
             self._do_request, method="POST", url=url, json=body
         )
 
-    def _do_unassociate(self, repo_id, criteria=None):
+    def _do_unassociate(self, repo_id, criteria=None, limit=None):
```

Comment:
After some discussion on private chat -> I think you will likely need also provide `skip` parameter in order to make unassociate to work with units outside the limit (aka first 100 units that match the criteria).

Reply:
I've opened a WIP PR for pubtools-pulp to show my general idea. It would continue sending unassociate requests sequentially until the number of units removed by a request is less than the limit. As we were discussing, this depends on  implementation detailsof both the python and pulp tasks

Reply:
How would `skip` be used?

I think it's not needed and the way Aidan is doing it is right, just loop and ask for a certain amount to be removed until there's none left.

If the idea is that `skip` would be used to submit multiple remove tasks concurrently, the naive approach to that wouldn't work.  limit=X skip=Y means "skip the first Y elements and then remove up to X elements".  If you were to queue e.g. limit=1000 skip=0, and limit=1000 skip=1000, Pulp will still process them sequentially, and it's not going to delete the first 2000 elements; it'll delete the first 1000, then leave a gap of 1000 NOT deleted, then delete the next 1000.  Just like how deleting elements from a list while iterating over it often causes similar bugs.